hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -104,7 +107,8 @@
     public static final String SERVICE_NAME = "hz:core:clientEngine";
 
     private static final int EXECUTOR_QUEUE_CAPACITY_PER_CORE = 100000;
-    private static final int THREADS_PER_CORE = 20;
+    private static final int BLOCKING_THREADS_PER_CORE = 20;
```

Comment:
is this too many?

Reply:
Since the threads can block the executor, for the functionality of clients we need more threads per core. 
The problem was exactly this multiplier before. We are trying to reduce number of message tasks that does blocking IO. The `THREADS_PER_CORE` was 20 , we switched to 1 because it is not needed on that executor anymore. 
`BLOCKING_THREADS_PER_CORE` has to remain high to be able to handle blocking tasks.